cpp/src_prims/sparse/knn.cuh

```diff
@@ -392,6 +392,27 @@ class sparse_knn_t {
       case ML::MetricType::METRIC_L2:
         pw_metric = raft::distance::DistanceType::L2Expanded;
         break;
+      case ML::MetricType::METRIC_L1:
+        pw_metric = raft::distance::DistanceType::L1;
+        break;
+      case ML::MetricType::METRIC_Canberra:
+        pw_metric = raft::distance::DistanceType::Canberra;
+        break;
+      case ML::MetricType::METRIC_Linf:
+        pw_metric = raft::distance::DistanceType::Linf;
+        break;
+      case ML::MetricType::METRIC_Lp:
+        pw_metric = raft::distance::DistanceType::LpUnexpanded;
```

Comment:
Probably I'm just misunderstanding, but it seems that "METRIC_L2" -> "L2Expanded" and "METRIC_Lp" -> "LpUnexpanded". Why the difference in expansion type?